backend/announcements/views.py

```diff
@@ -0,0 +1,27 @@
+from announcements.models import Announcement
+from announcements.permissions import AnnouncementPermissions
+from announcements.serializers import AnnouncementSerializer
+from django.db.models import Q
+from django.utils import timezone
+from rest_framework import viewsets
+
+
+class AnnouncementsViewSet(viewsets.ModelViewSet):
+    serializer_class = AnnouncementSerializer
+    permission_classes = [AnnouncementPermissions]
+
+    def get_queryset(self):
+        # automatically filter for active announcements
+        queryset = Announcement.objects.filter(
+            Q(release_time__lte=timezone.now())
+            & (Q(end_time__gte=timezone.now()) | Q(end_time__isnull=True))
+        ).prefetch_related("audiences")
+        audiences = self.request.query_params.get("audience")
+
+        if audiences:
+            audience_names = audiences.split(",")
+            queryset = queryset.filter(
+                audiences__name__in=[name.strip().upper() for name in audience_names]
```

Comment:
Also for reference if performance is an issue (I don't think it is for announcements lol) but you can use `icontains` to avoid an `upper` call, and you can also do a batch `Q` query using the `|=` operator:
```
q = Q()
for name in audience_names:
    q |= Q(audiences__name__icontains=name.strip())
queryset = queryset.filter(q)
```